ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -817,13 +817,19 @@ def runTest(self):
             self.log("Starting reachability state watch thread...")
             self.watching    = True
             self.light_probe = False
+            self.watch_dataplane = True # Flag for reachability_watcher.
+            self.watch_controlplane = True # Flag for reachability_watcher.
+            self.watcher_dp_is_stopped = threading.Event() # Waiter Event for the Watcher of dataplane state is stopped.
+            self.watcher_dp_is_running = threading.Event() # Waiter Event for the Watcher of dataplane state is running.
+            self.watcher_cp_is_stopped = threading.Event() # Waiter Event for the Watcher of controlplane state is stopped.
+            self.watcher_cp_is_running = threading.Event() # Waiter Event for the Watcher of control state is running.
+            self.watcher_dp_is_stopped.set()               # By default the Watchers are not running.
+            self.watcher_cp_is_stopped.set()
+            self.watcher_dp_is_running.clear()             # By default its required to wait for the Watchers started.
+            self.watcher_cp_is_running.clear()
             watcher = pool.apply_async(self.reachability_watcher)
-            self.watcher_is_stopped = threading.Event() # Waiter Event for the Watcher state is stopped.
-            self.watcher_is_running = threading.Event() # Waiter Event for the Watcher state is running.
-            self.watcher_is_stopped.set()               # By default the Watcher is not running.
-            self.watcher_is_running.clear()             # By default its required to wait for the Watcher started.
-            # Give watch thread some time to wind up
-            time.sleep(5)
+            self.watcher_dp_is_running.wait(5)
+            self.watcher_cp_is_running.wait(5)
```

Comment:
1. what we're waiting for here?
2. What if after 5 minutes the resources are not ready yet?

Reply:
1. We are waiting untill the reachability_watcher() enters running state, i.e. runs at least one _while_ iteration.
Its a replacement of _sleep(5)_.
2. _5_ is 5 seconds. Nothing happens when timeout is reached, the test goes further.
It just gives a time (max 5 seconds) for the reachability_watcher() to wind up.
Without the wait timeout, if reachability_watcher() for some reason failed to start, wait() will wait forever and hang up.

Reply:
The wait is covered by time.sleep(5) already. And the following code would wait until the data plane and control plane becomes stable before proceeding with test. I don't think you need to complicate the control flow here.

Reply:
But if we wanted to have only controlplane watcher enabled (and disable dataplane watcher) during the warm-reboot case, this is the way how to have data/control watcher behave separately, in scope of single reachability_watcher.